### PR TITLE
Moves Culling.cpp::ProcessWorklist debug variables under preprocessing

### PR DESCRIPTION
**Pull request by Jackerty**
Function Culling.cpp::ProcessWorklist has a couple of variables which is unused when compiling a profile causing
a -Werror compile error. Fix is to move variables and code increment them under debugging ifdef.

NOTE: I did not test the code since there was another unused variable error. However, variables are local to the function so this change should not cause any problems.

Signed-off-by: Jackerty <contact.jackerty@gmail.com>

### CODE REVIEW

**Review by amzn-tommy — APPROVED** [review]

**Review by rgba16f — COMMENTED** [review]
nit: could you remove the unrelated whitespace changes.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 1]
```diff
@@ -387,8 +393,11 @@ namespace AZ
                                 if (TestOcclusionCulling(worklistData, visibleEntry) == MaskedOcclusionCulling::CullingResult::VISIBLE)
 #endif
                                 {
-                                    numDrawPackets += AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
-                                    ++numVisibleCullables;
+                                    #ifdef AZ_CULL_DEBUG_ENABLED
+                                        ++numVisibleCullables;
+                                        numDrawPackets += // Return value of AddLodDataToView
+                                    #endif
+                                    AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
```
[the remark this answers is not in the document]
Comment (amzn-sj) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:400]: We could also just add an else block:
```
#if AZ_CULL_DEBUG_ENABLED
    numDrawPackets += AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, 
        *worklistData->m_view);
    ++numVisibleCullables;
#else
    AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
#endif
```

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 2]
```diff
@@ -267,7 +267,7 @@ namespace AZ
             return m_visScene->GetEntryCount();
         }
 
-
+
```
Comment (amzn-sj) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:270]: Formatting nit: Looks like extra spaces here.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 3]
```diff
@@ -296,13 +296,13 @@ namespace AZ
 #endif
             return worklistData;
         }
-
+
```
Comment (amzn-sj) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:299]: Formatting nit: Looks like extra spaces here.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 4]
```diff
@@ -898,7 +907,7 @@ namespace AZ
                         {
                             const Matrix4x4& worldToClip = viewPtr->GetWorldToClipMatrix();
                             Frustum frustum = Frustum::CreateFromMatrixColumnMajor(worldToClip, Frustum::ReverseDepth::True);
-                            m_debugCtx.m_frozenFrustums.insert({ viewPtr.get(), frustum });
+                            m_debugCtx.m_frozenFrustums.insert({ viewPtr.get(), frustum });
```
Comment (amzn-sj) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:910]: Formatting nit: Looks like extra spaces here.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 5]
```diff
@@ -911,7 +920,7 @@ namespace AZ
         }
 
         void CullingScene::EndCulling()
-        {
+        {
```
Comment (amzn-sj) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:923]: Formatting nit: Looks like extra spaces here.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 6]
```diff
@@ -527,9 +536,9 @@ namespace AZ
 #endif
 
         void CullingScene::ProcessCullablesCommon(
-            const Scene& scene [[maybe_unused]],
-            View& view,
-            AZ::Frustum& frustum [[maybe_unused]],
+            const Scene& scene [[maybe_unused]], 
+            View& view, 
+            AZ::Frustum& frustum [[maybe_unused]],
```
Comment (amzn-sj) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:541]: Formatting nit: Looks like extra spaces here.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgba16f) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:400]: > We could also just add an else block:
> 
> ```
> #if AZ_CULL_DEBUG_ENABLED
>     numDrawPackets += AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, 
>         *worklistData->m_view);
>     ++numVisibleCullables;
> #else
>     AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
> #endif
> ```

The else block is brittle because it requires 2 copies of the function call to be updated any time it is changed.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 7]
```diff
@@ -296,13 +296,13 @@ namespace AZ
...
         }
-
+            
         constexpr size_t WorkListCapacity = 5;
         using WorkListType = AZStd::fixed_vector<AzFramework::IVisibilityScene::NodeData, WorkListCapacity>;
 
 #if AZ_TRAIT_MASKED_OCCLUSION_CULLING_SUPPORTED
         static MaskedOcclusionCulling::CullingResult TestOcclusionCulling(
-                    const AZStd::shared_ptr<WorklistData>& worklistData,
+                    const AZStd::shared_ptr<WorklistData>& worklistData,
```
Comment (spham-amzn) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:305]: Related to the extra spaces above, you can reduce the diff of this change overall by removing the unnecessary whitespace differences

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jackerty) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:400]: Yeah, I didn't do #else because the annoying bug possibility was too high.
I change to rgba16f proposal on this with a comment that maybe_unused does not need to be fixed.

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp (hunk 7 above)
Comment (Jackerty) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:305]: Sorry about these. I think rebase caused them.
I will be more careful with the next PR I make.

**Review by rgba16f — APPROVED** [review]

**Comment by rgba16f** [comment]
Started an AR run for this change - https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-7043/

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp (hunk 7 above)
Comment (spham-amzn) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp]: Sometimes there are settings in the IDE to auto-format code

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 8]
```diff
@@ -351,8 +354,15 @@ namespace AZ
                                 if (TestOcclusionCulling(worklistData, visibleEntry) == MaskedOcclusionCulling::CullingResult::VISIBLE)
 #endif
                                 {
-                                    numDrawPackets += AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
-                                    ++numVisibleCullables;
+                                    // There are ways to write this without [[maybe_unused]], but they are brittle.
+                                    // For example, using #else could cause a bug where the function's parameter
+                                    // is changed in #ifdef but not in #else.
+                                    [[maybe_unused]] const uint32_t drawPacketCount=AddLodDataToView(c->m_cullData.m_boundingSphere.GetCenter(), c->m_lodData, *worklistData->m_view);
```
Comment (spham-amzn) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:360]: very minor: should be a space around '=' (drawPacketCount = AddLodDataToView)

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 9]
```diff
@@ -387,8 +397,15 @@ namespace AZ
...
```
Comment (spham-amzn) [inline Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:403]: very minor: should be a space around '=' (drawPacketCount = AddLodDataToView)
